## Exploratory Data Analysis of Iris Dataset

<a href="#Bottom">Bottom</a>
<p id="Top"></p>

In [16]:
import numpy as np
import pandas as pd

In [19]:
data = pd.read_csv(r"E:\Machine Learning\NoteBooks\ML\AIML\Data Git\SDP\Weekly challenges\Data\Iris\Iris.csv")

In [20]:
data.head()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa

In [22]:
data.tail()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm  \
145  146            6.7           3.0            5.2           2.3   
146  147            6.3           2.5            5.0           1.9   
147  148            6.5           3.0            5.2           2.0   
148  149            6.2           3.4            5.4           2.3   
149  150            5.9           3.0            5.1           1.8   

            Species  
145  Iris-virginica  
146  Iris-virginica  
147  Iris-virginica  
148  Iris-virginica  
149  Iris-virginica

In [34]:
data.set_index("Id",inplace=True)

In [40]:
data.head()

SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
Id                                                                       
1             5.1           3.5            1.4           0.2  Iris-setosa
2             4.9           3.0            1.4           0.2  Iris-setosa
3             4.7           3.2            1.3           0.2  Iris-setosa
4             4.6           3.1            1.5           0.2  Iris-setosa
5             5.0           3.6            1.4           0.2  Iris-setosa

In [35]:
data.shape

(150, 5)

In [36]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 150 entries, 1 to 150
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   SepalLengthCm  150 non-null    float64
 1   SepalWidthCm   150 non-null    float64
 2   PetalLengthCm  150 non-null    float64
 3   PetalWidthCm   150 non-null    float64
 4   Species        150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 7.0+ KB


In [37]:
data.isna().sum()

SepalLengthCm    0
SepalWidthCm     0
PetalLengthCm    0
PetalWidthCm     0
Species          0
dtype: int64

- The Data is having no null values

In [69]:
data['Species'].value_counts()

Iris-versicolor    50
Iris-virginica     50
Iris-setosa        50
Name: Species, dtype: int64

In [71]:
data.groupby(data['Species']).count()

SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
Species                                                                  
Iris-setosa                 50            50             50            50
Iris-versicolor             50            50             50            50
Iris-virginica              50            50             50            50

- We have data of 50 flower from each species

In [39]:
data.describe()

SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
count     150.000000    150.000000     150.000000    150.000000
mean        5.843333      3.054000       3.758667      1.198667
std         0.828066      0.433594       1.764420      0.763161
min         4.300000      2.000000       1.000000      0.100000
25%         5.100000      2.800000       1.600000      0.300000
50%         5.800000      3.000000       4.350000      1.300000
75%         6.400000      3.300000       5.100000      1.800000
max         7.900000      4.400000       6.900000      2.500000

In [73]:
meanSL = data['SepalLengthCm'].mean() 
meanSW = data['SepalWidthCm'].mean()
meanPw = data['PetalWidthCm'].mean()
meanpl = data['PetalLengthCm'].mean()

In [89]:
a = data[data.SepalLengthCm > meanSL]
print('Number of Flower having greater SepalLengthCm than mean-:',a.shape[0])

Number of Flower having greater SepalLengthCm than mean-: 70


In [90]:
b = data[data['SepalWidthCm']>meanSW]
print('Number of Flower having greater SepalWidthCm than mean-:',b.shape[0])

Number of Flower having greater SepalWidthCm than mean-: 67


In [91]:
p = data[data['PetalWidthCm']>meanPw]
print('Number of Flower having greater PetalWidthCm than mean-:',p.shape[0])

Number of Flower having greater PetalWidthCm than mean-: 90


In [92]:
pl = data[data['PetalLengthCm']>meanpl]
print('Number of Flower having greater PetalLength than mean-:',pl.shape[0])

Number of Flower having greater PetalLength than mean-: 93


In [94]:
# Number of flower having petal length more than mean petal length and  flower having sepal length greater than mean sepal length
plsl = a[a['PetalLengthCm']>meanpl]
plsl

SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm          Species
Id                                                                            
51             7.0           3.2            4.7           1.4  Iris-versicolor
52             6.4           3.2            4.5           1.5  Iris-versicolor
53             6.9           3.1            4.9           1.5  Iris-versicolor
55             6.5           2.8            4.6           1.5  Iris-versicolor
57             6.3           3.3            4.7           1.6  Iris-versicolor
..             ...           ...            ...           ...              ...
146            6.7           3.0            5.2           2.3   Iris-virginica
147            6.3           2.5            5.0           1.9   Iris-virginica
148            6.5           3.0            5.2           2.0   Iris-virginica
149            6.2           3.4            5.4           2.3   Iris-virginica
150            5.9           3.0            5.1           1.8   Iris-virginica

[70 rows x 5 columns]

- The Above 70 Flowers are Bigger than flower present in rest of the dataset 
- Out of which 44 Belong to Iris-virginica and 26 Belong to Iris-versicolor

In [95]:
plsl['Species'].value_counts()

Iris-virginica     44
Iris-versicolor    26
Name: Species, dtype: int64

## Summary
- The Iris Data set is having 150 Flowers Data
- 50 of Iris-virginica
- 50 of Iris-versicolor
- 50 of Iris-Setosa
- Iris-virginica and versicolor are mainly bigger flower than iris-setosa
- Iris-virginica was having bigger Petal length and sepal length, Around 44 flower proves this result
- 70 Flowers have Sepal length Greater than mean sepal length
- 67 Flowers have Sepal Width Greater than mean Sepal width
- 90 Flowers have Petal Width Greater than mean Petal width
- 93 Flowers have Petal Length Greter than mean Petal length

<a href="#Top">Top</a>
<p id="Bottom"></p>